### PR TITLE
Add orientation indicator and improve orientation filtering

### PR DESCRIPTION
**Pull request by ElvinC**
Changes:

* Added an orientation indicator (hidden under checkbox) showing axes, integrated orientations, and smoothed camera orientations.
* Tweaked the complementary filter (Modified gains, added accel auto calibration, slight IIR filtering, modified dynamic gain curve) . Note that original ported version (with modified gains) is left in a separate file but currently unused.
* Added rust port of the offline [VQF orientation filter](https://github.com/dlaidig/vqf/tree/main/vqf/cpp) currently using the default parameters. Offline filtering can give better results in theory, though new complementary filter seems better for flight footage.

This solves #224

### CODE REVIEW

File: src/core/integration_complementary.rs [hunk 1]
```diff
@@ -413,4 +413,4 @@ fn rotate_vector_by_quaternion(x: f64, y: f64, z: f64, q0: f64, q1: f64, q2: f64
         2.0*(q1*q2 + q0*q3)*x + (q0*q0 - q1*q1 + q2*q2 - q3*q3)*y + 2.0*(q2*q3 - q0*q1)*z,
         2.0*(q1*q3 - q0*q2)*x + 2.0*(q2*q3 + q0*q1)*y + (q0*q0 - q1*q1 - q2*q2 + q3*q3)*z
     )
-}
+}
```
Comment (DusKing1) [inline src/core/integration_complementary.rs:416]: missing end of line

File: src/core/integration_vqf.rs [hunk 2]
```diff
@@ -0,0 +1,1313 @@
...
+        filter_delta(&mag_dist, n, ts, &params, true, delta); // backward direction
+        for i in 0..n  {
+           quat9d.splice(4*i..4*i+4, (&VQF::quat_apply_delta(quat6d[4*i..4*i+4].try_into().unwrap(), delta[i])).iter().cloned());
+        }
+    } else {
+        // delta = &mut vec![0.0; n];
+        // quat9d = &mut vec![0.0; 4*n];
+    }
+
+}
```
Comment (DusKing1) [inline src/core/integration_vqf.rs:1313]: missing EOL

File: src/core/integration.rs [hunk 3]
```diff
@@ -174,45 +176,95 @@ impl GyroIntegrator for GyroOnlyIntegrator {
...
+
+        // Tweak parameters here, see parameter descriptions: https://github.com/dlaidig/vqf/blob/main/vqf/cpp/vqf.hpp#L37
+        let params = VQFParams::default();
+        offline_vqf(gyr, acc, Some(mag), num_samples, sample_time, params, None, Some(&mut quat), None, None, None, None, None);
+        for (i, v) in imu_data.iter().enumerate() {
+            out_quats.insert((v.timestamp_ms * 1000.0) as i64, Quat64::from_quaternion(Quaternion::from_parts(quat[i*4], Vector3::new(quat[i*4+1], quat[i*4+2], quat[i*4+3]))));
+        }
+        out_quats
+    }
+}
```
Comment (DusKing1) [inline src/core/integration.rs:270]: missing EOL

File: src/ui/menu/MotionData.qml [hunk 4]
```diff
@@ -332,6 +333,110 @@ MenuItem {
...
+                            ctx.fill();
+                            ctx.stroke();
+                        }
+
+                        ctx.lineWidth = 1.5;
+                        ctx.strokeStyle = maincolor;
+                        ctx.globalAlpha = 0.8;
+                        ctx.lineJoin = "bevel";
+                        for (var view = 0; view < 2; view++) {
+                            for (var linenum = 0; linenum < lines.length; linenum++) {
```
Comment (AdrianEddy) [inline src/ui/menu/MotionData.qml:415]: var -> let

File: src/ui/menu/MotionData.qml [hunk 5]
```diff
@@ -332,6 +333,110 @@ MenuItem {
...
+                            ctx.arc(width/6*(i*2+1), height/2, 4, 0, 2 * Math.PI, false);
+                            ctx.fillStyle = maincolor;
+                            ctx.fill();
+                            ctx.stroke();
+                        }
+                        
+                        for (let i = 0; i < 3; i++) {
+                            ctx.beginPath();
+                            ctx.moveTo(width/6, height/2);
+                            let transformedvec = transform.times(vecs[i])
```
Comment (AdrianEddy) [inline src/ui/menu/MotionData.qml:396]: let -> const

File: src/ui/menu/MotionData.qml [hunk 6]
```diff
@@ -332,6 +333,110 @@ MenuItem {
...
+                                    [2,4,3]]
+                        let cam_vert_vecs = []
+                        for (var i = 0; i < cam_vertices.length; i++) {
+                            cam_vert_vecs.push(Qt.vector3d(cam_vertices[i][0],cam_vertices[i][1],cam_vertices[i][2]));
+                        }
+                        let quats = controller.quats_at_timestamp(Math.round(currentTimestamp))
+                        let transform = Qt.quaternion( quats[0], quats[1],  quats[2], quats[3]); // wxyz
+                        let maincolor = style === "light" ? "rgba(0,0,0,0.9)" : "rgba(255,255,255,0.9)";
+                        let transform_smooth = transform.times(Qt.quaternion( quats[4], quats[5],  quats[6], quats[7]).inverted());
+                        let transforms = [transform, transform_smooth]
```
Comment (AdrianEddy) [inline src/ui/menu/MotionData.qml:382]: If I see correctly none of these changes so use `const`

File: src/ui/VideoArea.qml [hunk 7]
```diff
@@ -265,6 +265,7 @@ Item {
                 onCurrentFrameChanged: {
                     fovChanged();
                     controller.update_keyframe_values(timestamp);
+                    window.motionData.orientationIndicator.updateOrientation(timeline.position * timeline.durationMs * 1000)
```
Comment (AdrianEddy) [inline src/ui/VideoArea.qml:268]: isn't `timeline.position * timeline.durationMs * 1000` the same as just `timestamp`?

Comment (ElvinC) [reply src/ui/VideoArea.qml:268]: Oh oops, just `timestamp*1000` works the same. Somehow missed that one

**Pull request merged**